website/docs: prepare release notes for 2024.8 (#11011)
goauthentik/authentik: website/docs/releases/2024/v2024.8.md

To try out the release candidate, replace your Docker image tag with the latest release candidate number, such as 2024.8.0-rc1. You can find the latest one in [the latest releases on GitHub](https://github.com/goauthentik/authentik/releases). If you don't find any, it means we haven't released one yet.
:::::

## Highlights

> -   **Source property mappings for SCIM, OAuth, SAML and Plex sources**: All sources now support property mappings to configure how authentik interprets the data, and groups can be synced from all sources that provide group info.

Reviewer: ```suggestion
-   **Source property mappings for SCIM, OAuth, SAML and Plex sources**: All sources now support property mappings to configure how authentik interprets the incoming source's data. Additionally, groups can be synced from all sources that provide group info.
```

> -   **RADIUS provider custom attribute support**: Integrate RADIUS better into network infrastructure by sending vendor-specific attributes based on the user authenticating.

Reviewer: ```suggestion
-   **RADIUS provider custom attribute support**: Integrate RADIUS better into network infrastructure by sending vendor-specific attributes based on the user who is authenticating.
```

Reviewer: How can we best clarify what "sending" means? To where is the vendor-specific attributes being sent? To the local network infrastructure?

Author: I dont think it needs clarification, the attributes are being sent to whatever connects to the Radius outpost

> -   **SAML encryption support**: SAML source and provider support encryption now, which prevents anyone from viewing the information of Assertions in-flight.

Reviewer: ```suggestion
-   **SAML encryption support**: SAML source and provider data is now encrypted, which prevents anyone from viewing the information of Assertions in-flight.
```

Author: This is an optional feature so saying "data is now encrypted" is not true

> -   **RBAC support for Blueprints and Terraform**: Permissions can now be assigned and automated using both Blueprints and Terraform.

Reviewer: Will we have the Terraform process documented in time for the release?

-   **Source property mappings for SCIM, OAuth, SAML and Plex sources**

>     All source types now support Property mappings to customize how authentik should interpret the data the source provides. In addition to that, it is also now possible to sync groups and group membership from sources that provide group information. See [Property Mappings](../../sources/property-mappings/index.md).

Reviewer: ```suggestion
    All source types now support property mappings to customize how authentik should interpret the data the source provides. In addition to that, it is also now possible to sync groups and group membership from sources that provide group information. See [Property Mappings](../../sources/property-mappings/index.md).
```

-   **RADIUS provider custom attribute support**

    With 2024.8 it is possible to define custom attributes for the RADIUS provider, for example vendor specific attributes like Cisco's `AV-Pair` attribute. These attributes are defined in property mappings which means they can be dynamically defined based on the user authenticating. See [RADIUS Provider](../../providers/radius/index.mdx#radius-attributes)

-   **SAML encryption support**

>     It is now possible to configure a SAML Source/Provider to decrypt and validate encrypted assertions. This can be configured by certain a [Certificate-keypair](../../core/certificates.md) and selecting it in the SAML Source/Provider.

Reviewer: ```suggestion
    It is now possible to configure a SAML source and/or provider to decrypt and validate encrypted assertions. This can be configured by creating a [Certificate-keypair](../../core/certificates.md) and selecting it in the SAML source os provider.
```

Author: ```suggestion
It is now possible to configure a SAML sources and providers to decrypt and validate encrypted assertions. This can be configured by creating a [Certificate-keypair](../../core/certificates.md) and selecting it in the SAML source or provider.
```

-   **GeoIP Policy**

>     With the new [GeoIP Policy](../../policies/index.md#geoip-policy) it is possible to grant/deny access based on Country and ASN, without having to write an expression.

Reviewer: ```suggestion
    With the new [GeoIP Policy](../../policies/index.md#geoip-policy) it is possible to grant/deny access based on Country and ASN, without having to write an expression policy.
```

-   **Simplification of LDAP Provider permissions**

>     The LDAP provider now uses RBAC to assign the permission to search the full directory instead of requiring a dedicated group to be created. As part of the upgrade, existing search groups' users are migrated to grant the required permission to search the full directory.

Reviewer: ```suggestion
    The LDAP provider now uses RBAC to assign the permission to search the full directory instead of requiring a dedicated group to be created. As part of the upgrade, existing search groups' users are migrated into the RBAC-managed group in order to grant the required permission to search the full directory.
```

Reviewer: @Jens I am not sure about my rewording. During th eupgrade, are the users in legacy, existing search groups migrated into a new group...? Or how exactly do each of those users get granted the correct permissions to search? Maybe they get assigned a role...? Needs a little more detail here.

Author: The users in the group are directly assigned the permission via user-bound object-specific permissions, no new group or role is created

 - web: restore hasLaunchUrl to client-side criteria for filtering apps (#10291)
 - web: search select with focus, autocomplete, and progressive search (#10728)
 - web: set noopener and noreferrer on all external links (#10304)
 - web: set noopener and noreferrer on all external links (#10304)
 - web: update to ESLint 9 (#10812)
 - web/admin: fix access token list calling wrong API (#10434)
 - web/admin: fix authentication/enrollment flow in sources being marked as required (#10911)
 - web/admin: fix missing SAML Provider ECDSA options (#10612)
 - web/admin: fix selectable card colour in dark theme (#10794)
 - web/admin: refactor property mappings forms (#10810)
 - web/admin: show matching user reputation scores in user details (#10276)
 - web/admin: widen prompt form (#10615)
 - web/elements: fix empty enterprise banner (#10882)
 - web/flows: clean up loading, syntax and transitions (#10792)
 - web/flows: don't grab focus for password input on identification stage (#10593)
 - web/flows: remove background image link (#10318)
 - web/flows: remove continue button from AutoSubmit stage (#10253)
 - web/flows: Simplified flow executor (#10296)

>  - web/flows: Simplified flow executor (#10296)

Reviewer 2: weird duplication

Author: yeah this is partially caused by how we generate that list, comparing `main` to `version-2024.6` which has some commits cherry picked on it
